app/containers/Ico.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { IcoPhase } from "../actions/constants";
+import { loadIcoParams } from "../actions/loadIcoParams";
+import { selectIcoPhase } from "../reducers/icoParameters";
+import BeforeIco from "./BeforeIco";
+import DuringIco from "./DuringIco";
+
+interface IcoProps {
+  icoPhase: IcoPhase;
+  loadIcoParameters: any;
+}
+
+export const Ico: React.SFC<IcoProps> = props => {
+  const { icoPhase } = props;
+  props.loadIcoParameters();
+  // @todo: this should load after did mount
+  switch (icoPhase) {
```

Comment:
probably icoPhase and APP_MODE from env variables should be merged into one entity